openlibrary/plugins/worksearch/code.py

```diff
@@ -415,6 +415,16 @@ def get_facet(facets, f, limit=None):
 re_olid = re.compile('^OL\d+([AMW])$')
 olid_urls = {'A': 'authors', 'M': 'books', 'W': 'works'}
 
+class scan(delegate.page):
+    """
+    Experimental EAN barcode scanner page to scan and add/view books by their barcodes.
+    """
+    path = "/barcodescanner"
+    def GET(self):
+        page = render.barcodescanner()
```

Comment:
Probably not what we want to do re: keeping our templates js-free, but one possible "hack" may be to use this as a way to pass in the cover server and include a `<script> .... coverurl=$()</script>`